### PR TITLE
fix: Update the Unix render to use the new API + interactive traces

### PR DESCRIPTION
**Pull request by mmcqd**
* Updates `Asai_unix` and the `STLC` examples to use the new API
* Fixes a bug where segments could contain newlines
* Adds a new `interactive_trace` function to `Asai_unix`, which dumps the user into a terminal app where they can move up and down the stack of traces using arrow keys

### CODE REVIEW

File: src/unix/Asai_unix.ml [hunk 1]
```diff
@@ -1,82 +1,115 @@
...
 
-
-module Make (ErrorCode : ErrorCode.S) =
+module Make (Code : Code.S) =
 struct
-  module Diagnostic = Diagnostic.Make(ErrorCode)
-
   let vline ~attr height str =
     I.vcat @@ List.init height (fun _ -> I.string ~attr str)
+  let column ~align images =
```
Comment (TOTBWF) [inline src/unix/Asai_unix.ml:12]: I don't think we need to compute the width for line numbers here, shouldn't `<|>` and  `vcat` work?

File: examples/stlc/STLC.ml [hunk 2]
```diff
@@ -110,52 +109,53 @@ struct
...
     let lexbuf = Lexing.from_channel (open_in filepath) in
     lexbuf.lex_curr_p <- { lexbuf.lex_curr_p with pos_fname = filepath };
-    let contents = 
-      let ch = open_in filepath in
-      let str = really_input_string ch (in_channel_length ch) in
-      close_in ch;
-      str
-    in
-    Doctor.load_file ~filepath @@ contents;
+    (* let contents =
```
Comment (TOTBWF) [inline examples/stlc/STLC.ml:120]: We should delete this code if we aren't using it

**Review by TOTBWF — CHANGES_REQUESTED** [review]
Some minor nits, but otherwise this looks good!

File: src/unix/Asai_unix.ml (hunk 1 above)
Comment (mmcqd) [reply src/unix/Asai_unix.ml:12]: We need it for right aligned line numbers

Comment (TOTBWF) [reply src/unix/Asai_unix.ml:12]: Ah, make sense!

**Review by TOTBWF — APPROVED** [review]

**Pull request merged**